### PR TITLE
Import ottavas from Sibelius correctly

### PR DESCRIPTION
**Pull request by miiizen**
Sibelius exports its ottavas inconsistently with other programs - ottava tags should start before the notes under the ottava, and finish *after* the last note.  Sibelius stops before, leaving the ottava a note short.  This PR delays adding ottavas until the following note if the score was exported from Sibelius.
Before:
Screenshot 2024-02-13 at 15 14 10

After:
Screenshot 2024-02-13 at 15 05 24

(Depends on the commit identiying exporter software 2978b7b in #21483)

### CODE REVIEW

File: src/importexport/musicxml/internal/musicxml/importmxmlpass1.cpp [hunk 1]
```diff
@@ -1198,7 +1198,7 @@ void MusicXMLParserPass1::identification()
             // TODO
             while (m_e.readNextStartElement()) {
                 if (m_e.name() == "software") {
-                    m_exporterString += m_e.readText();
+                    m_exporterString += m_e.readText().toLower();
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass1.cpp:1201]: This doesn't mix and match with
```cpp
            if (_pass1.exporterString().contains("Dolet")) {
```

**Review by oktophonie — APPROVED** [review]

**Review by mike-spa — APPROVED** [review]

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.h [hunk 2]
```diff
@@ -411,6 +412,7 @@ class MusicXMLParserDirection
     String m_sndFine;
     String m_sndSegno;
     String m_sndToCoda;
+    bool visible = true;
```
Comment (cbjeukendrup) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.h:419]: This seems unused

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 3]
```diff
@@ -3175,6 +3183,59 @@ void MusicXMLParserDirection::dynamics()
...
+        std::map<String, String> otherDirectionStrings;
+        if (m_pass1.exporterString().contains(u"Dolet")) {
+            otherDirectionStrings = {
+                { String(u"To Coda"), String(u"To Coda") },
+                { String(u"Segno"), String(u"<sym>segno</sym>") },
+                { String(u"CODA"), String(u"<sym>coda</sym>") },
+            };
+        }
+        std::map<String, SymId> otherDirectionSyms;
+        otherDirectionSyms = { { String(u"Rhythm dot"), SymId::augmentationDot },
```
Comment (cbjeukendrup) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:3233]: For now, maybe nicer to do this one like this:
```suggestion
        static const std::map<String, SymId> otherDirectionSyms = { { String(u"Rhythm dot"), SymId::augmentationDot },
```

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 4]
```diff
@@ -3002,7 +3013,13 @@ void MusicXMLParserDirection::direction(const String& partId,
             m_logger->logError(u"spanner already stopped", &m_e);
             delete desc.sp;
         } else {
-            if (spdesc.isStarted) {
+            if (spdesc.sp && spdesc.sp->isOttava() && delayOttava) {
```
Comment (cbjeukendrup) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:3016]: Is it correct that you don't need to check `spdesc.isStarted` here?

**Comment by cbjeukendrup** [comment]
And a rebase is needed

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp (hunk 4 above)
Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:3016]: It is the else branch of `if (spdesc._isStopped)`, can it have a 3rd state?

Comment (miiizen) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:3016]: @cbjeukendrup I probably ought to, it can't hurt to be safe

**Review by cbjeukendrup — APPROVED** [review]

**Pull request merged**